### PR TITLE
XOR Linked List

### PR DESCRIPTION
**Pull request by radhikaaojha**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

XOR Linked List implementation in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#7088

### CODE REVIEW

**Comment by radhikaaojha** [comment]
Please review my PR @ankitaggarwal23

**Comment by radhikaaojha** [comment]
@atarax665

File: C/README.md [hunk 1]
```diff
@@ -134,6 +134,7 @@
 - [Reverse Queue using stack](ds/reverse-queue.c)
 - [Postfix to Infix Conversion](ds/postfixToInfix.c)
 - [Prefix to Postfix Conversion](ds/prefix-to-postfix.c)
+- [XOR Linked List ] (ds/XOR.c)
```
Comment (ankitaggarwal23) [inline C/README.md]: Remove the space after `List` and also between` ] (`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by radhikaaojha** [comment]
I have updated the changes requested @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**